packages/pharos/src/components/toast/pharos-toaster.ts

```diff
@@ -1,32 +1,83 @@
 import { PharosElement } from '../base/pharos-element';
 import { html } from 'lit';
 import type { TemplateResult, CSSResultArray } from 'lit';
+import { state } from 'lit/decorators.js';
+import { repeat } from 'lit/directives/repeat.js';
 import { toasterStyles } from './pharos-toaster.css';
 
-import type { PharosToast } from './pharos-toast';
+import type { ToastStatus } from './pharos-toast';
+import { PharosToast } from './pharos-toast';
 import { DEFAULT_STATUS, DEFAULT_INDEFINITE } from './pharos-toast';
 
 import { v4 as uuidv4 } from 'uuid';
+import ScopedRegistryMixin from '../../utils/mixins/scoped-registry';
 
 /**
  * pharos-toast-open event.
  *
- * @tag pharos-toaster
- *
  * @event pharos-toast-open
- * @type {object}
- * @property {ToastStatus} status - The status for the toast.
- * @property {string} content - The content to slot into the toast.
+ * @type {ToastCreateDetail}
+ */
+
+/**
+ * pharos-toast-update event.
+ *
+ * @event pharos-toast-update
+ * @type {ToastUpdateDetail}
+ */
+
+/**
+ * pharos-toast-close event.
+ *
+ * @event pharos-toast-close
+ * @type {ToastCloseDetail}
  */
 
+type ToastID = string;
+type ToastContent = string;
+type ToastIndefinite = boolean;
+
+interface ToastDetail {
+  id: ToastID;
+  content: ToastContent;
+  status: ToastStatus;
+  indefinite: ToastIndefinite;
+}
+
+interface ToastCreateDetail {
+  id?: ToastID;
+  content: ToastContent;
+  status?: ToastStatus;
+  indefinite?: ToastIndefinite;
+}
+
+interface ToastUpdateDetail {
+  id: ToastID;
+  content?: ToastContent;
+  status?: ToastStatus;
+}
+
+interface ToastCloseDetail {
+  id: ToastID;
+}
```

Comment:
Open to toning this down or suggestions on where to move these—we don't have an established pattern. I could see there being a good case for something like `src/components/<comp>/pharos-<comp>.types.ts`.

Reply:
Personally like to keep types close to where they are used. We could consider moving them out into their own module when we need to use them elsewhere or wish to expose them to the consumer?